File: 2.simple string_checking/abhi.c
    #include<stdio.h> 
    #include<string.h>
    void main() {
    	char string[50];
    	int flag,count=0;
    	printf("The grammar is: S->aS,S->E, S->Sb, S->ab\n");
    	printf("Enter the string to be checked:\n");
	scanf("%s",string);
	if(string[0]=='b')
		printf("string does not belong to specified grammar");
    	else if(string[0]=='a') {
    		flag=0;
    		for (count=1;string[count-1]!='\0';count++) {
    			if(string[count]=='b') {
    				flag=1;
    				continue;
    			} else if((flag==1)&&(string[count]=='a')) {
    				printf("The string does not belong to the specified grammar");
    				break;
    			} else if(string[count]=='a')
    			continue;
    			else {
    				printf("String  accepted");
    			}
    		}
    	}
    	
    }

File: 2.simple string_checking/abhi.py
print("Grammar : S->aS|Sb|ab|E")
string = str(input('Enter a String : '))
false=0
if(string[0]=='a'):
	flag=0
	for c in string[1:]:
		if(c=='b'):	
			flag=1
			continue
		elif((flag==1) and (c=='a')): 
			print("The string does not belong to the specified grammar")
			false=1
			break
		elif(c=='a'):
			continue
		
	if false==0:
		print("String accepted")
